src/runtime.c

```diff
@@ -379,6 +380,14 @@ main (int argc, char *argv[])
         if ((err = sqfs_open_image(&fs, appimage_path, fs_offset)))
             exit(1);
 
+        // track duplicate inodes for hardlinks
+        created_inode = malloc(fs.sb.inodes * sizeof(char *));
+        if(created_inode != NULL) {
+            memset(created_inode, 0, fs.sb.inodes * sizeof(char *));
+        } else {
+            fprintf(stderr, "Can't track hardlinks.\n");
```

Comment:
I just saw your question. Yes, if that isn't possible, I'd rather exit instead of falling back to creating copies. This way, the users can report the issues to us, and we can hopefully fix them. I don't like implicit fallbacks.